src/test/java/com/hankcs/hanlp/utility/MathUtilityTest.java

```diff
@@ -0,0 +1,95 @@
+package com.hankcs.hanlp.utility;
+
+import com.hankcs.hanlp.corpus.tag.Nature;
+import com.hankcs.hanlp.dictionary.CoreDictionary;
+import com.hankcs.hanlp.seg.common.Vertex;
+import org.junit.Assert;
+import org.junit.Test;
+import java.util.HashMap;
+
+public class MathUtilityTest {
+    static final double DELTA = 0.0;
+
+    @Test
+    public void testAverage() {
+        Assert.assertEquals(
+            2.0,
+            MathUtility.average(new double[]{1.0, 2.0, 3.0}),
+            DELTA
+        );
+    }
+
+    @Test
+    public void testNormalizeExpDoubleArray() {
+        double[] predictionScores = {0, 1, 2, 3};
+        double[] expected = {
+            0.03205860328008499, 0.08714431874203257,
+            0.23688281808991013, 0.6439142598879724
+        };
+
+        MathUtility.normalizeExp(predictionScores);
+
+        Assert.assertArrayEquals(expected, predictionScores, DELTA);
+    }
+
+    @Test
+    public void testNormalizeExpHashMap() {
+        HashMap<String, Double> predictionScores =
+            new HashMap<String, Double>();
```

Comment:
You don't need the second` HashMap<String, Double>`. 
You can do it like this `HashMap<String, Double> predictionScores = new HashMap<>();`

Reply:
You can usually but this is a java 7+ feature but unfortunately this project is running java 6

Reply:
ok :)